scripts/ssl_tts/make_supdata.py

```diff
@@ -0,0 +1,434 @@
+
+    args = parser.parse_args()
+
+    device = "cuda"
+    manifest_path = args.manifest_path
+    ssl_model_ckpt_path = args.ssl_model_ckpt_path
+
+    dataset = AudioDataset(manifest_path)
+    dataloader = torch.utils.data.DataLoader(
+        dataset,
+        batch_size=args.batch_size,
+        shuffle=False,
+        collate_fn=dataset.pad_collate_fn,
+        num_workers=args.num_workers,
+    )
+
+    ssl_model = ssl_tts.SSLDisentangler.load_from_checkpoint(ssl_model_ckpt_path, strict=False)
+    with open_dict(ssl_model.cfg):
+        ssl_model.cfg.preprocessor.exact_pad = True
+    ssl_model.preprocessor = hydra.utils.instantiate(ssl_model.cfg.preprocessor)
+    ssl_model.preprocessor_disentangler = ssl_model.preprocessor
+    ssl_model.eval()
+    ssl_model.to(device)
+
+    fb = (
+        torch.tensor(librosa.filters.mel(sr=22050, n_fft=1024, n_mels=80, fmin=0, fmax=8000), dtype=torch.float,)
+        .unsqueeze(0)
+        .to(device)
+    )
+
+    st = time.time()
+    bidx = 0
+    wav_and_id_list = []
+
+    for batch in tqdm(dataloader):
+        bidx += 1
+        with torch.no_grad():
+            (
+                _,
+                _,
+                batch_content_embedding,
+                batch_content_log_probs,
+                batch_encoded_len,
+            ) = ssl_model.forward_for_export(
+                input_signal=batch['audio'].to(device),
+                input_signal_length=batch['audio_len'].to(device),
+                normalize_content=True,
+            )
+
+            batch_mel_specs = get_mel_spectrogram(fb, batch['audio'][:, :-1].to(device))
+            audio_segmented, segment_indices = segment_batch(batch)
+            audio_seg_len = torch.tensor([len(segment) for segment in audio_segmented]).to(device).long()
+
+            _, batch_speaker_embeddings, _, _, _ = ssl_model.forward_for_export(
+                input_signal=audio_segmented.to(device), input_signal_length=audio_seg_len, normalize_content=True,
+            )
+
+            for idx in range(batch['audio'].shape[0]):
+                _speaker = batch['speaker'][idx].item()
+                wav_path = batch['wav_path'][idx]
+
+                wav_id = batch['rel_audio_path_as_text_id'][idx]
+                wav_and_id_list.append((wav_path, wav_id, _speaker))
+                content_embedding = batch_content_embedding[idx].detach()
+                content_log_probs = batch_content_log_probs[:, idx, :].detach()  # (content lob prob is (t, b, c))
+                encoded_len = batch_encoded_len[idx].detach()
+                content_embedding = content_embedding[: encoded_len.item()]
+                content_embedding = content_embedding.t()
+                content_log_probs = content_log_probs[: encoded_len.item()]
+                content_log_probs = content_log_probs.t()
+                content_probs = torch.exp(content_log_probs)
+
+                duration = torch.ones(content_embedding.shape[1]) * 4.0
```

Comment:
Use user-provided param? (And in a few spots below)

Reply:
added user-provided params